inc/class-wp-saml-auth.php

```diff
@@ -61,22 +61,39 @@ public function get_provider() {
 	 */
 	public function action_init() {
 
-		$simplesamlphp_path = self::get_option( 'simplesamlphp_autoload' );
-		if ( file_exists( $simplesamlphp_path ) ) {
-			require_once $simplesamlphp_path;
-		}
-
-		if ( ! class_exists( 'SimpleSAML_Auth_Simple' ) ) {
-			add_action( 'admin_notices', function() {
-				if ( current_user_can( 'manage_options' ) ) {
-					// Translators: Links to the WP SAML Auth plugin.
-					echo '<div class="message error"><p>' . wp_kses_post( sprintf( __( "WP SAML Auth wasn't able to find the <code>SimpleSAML_Auth_Simple</code> class. Please check the <code>simplesamlphp_autoload</code> configuration option, or <a href='%s'>visit the plugin page</a> for more information.", 'wp-saml-auth' ), 'https://wordpress.org/plugins/wp-saml-auth/' ) ) . '</p></div>';
-				}
-			});
-			return;
-		}
-
-		$this->provider = new SimpleSAML_Auth_Simple( self::get_option( 'auth_source' ) );
+		$connection_type = self::get_option( 'connection_type' );
+		if ( 'internal' === $connection_type ) {
+			$autoload = dirname( __DIR__ ) . '/vendor/autoload.php';
+			if ( file_exists( $autoload ) ) {
+				require_once $autoload;
+			}
+			if ( ! class_exists( 'OneLogin_Saml2_Auth' ) ) {
+				add_action( 'admin_notices', function() {
+					if ( current_user_can( 'manage_options' ) ) {
+						// Translators: Links to the WP SAML Auth plugin.
+						echo '<div class="message error"><p>' . wp_kses_post( sprintf( __( "WP SAML Auth wasn't able to find the <code>OneLogin_Saml2_Auth</code> class. Please verify your Composer autoloader, or <a href='%s'>visit the plugin page</a> for more information.", 'wp-saml-auth' ), 'https://wordpress.org/plugins/wp-saml-auth/' ) ) . '</p></div>';
+					}
+				});
+				return;
+			}
+			$auth_config = self::get_option( 'internal_config' );
+			$this->provider = new OneLogin_Saml2_Auth( $auth_config );
+		} else {
+			$simplesamlphp_path = self::get_option( 'simplesamlphp_autoload' );
+			if ( file_exists( $simplesamlphp_path ) ) {
+				require_once $simplesamlphp_path;
+			}
+			if ( ! class_exists( 'SimpleSAML_Auth_Simple' ) ) {
+				add_action( 'admin_notices', function() {
+					if ( current_user_can( 'manage_options' ) ) {
+						// Translators: Links to the WP SAML Auth plugin.
+						echo '<div class="message error"><p>' . wp_kses_post( sprintf( __( "WP SAML Auth wasn't able to find the <code>SimpleSAML_Auth_Simple</code> class. Please check the <code>simplesamlphp_autoload</code> configuration option, or <a href='%s'>visit the plugin page</a> for more information.", 'wp-saml-auth' ), 'https://wordpress.org/plugins/wp-saml-auth/' ) ) . '</p></div>';
```

Comment:
Because I've introduced a new `connection_type=>(internal|simplesamlphp)` configuration argument, the default connection type still needs to be `simplesamlphp` to avoid a backwards compatibility break on existing installs.

The code path is such that you'll see one error message with one connection type, and the other with the other connection type. You must've had `connection_type=simplesamlphp` at the time.